### ISSUE TITLE
React-ify Find-Form Download Links Across Page Types

### ISSUE THREAD

**Issue by dsasser** [department-of-veterans-affairs/va.gov-cms#10851]
## Description

Move to a single react component for Find-A-Form PDF download links. This will reduce errors, complexity, and technical debt.

Veterans have two ways to download their Find-A-Form form assets (PDFs):
- From the Find-A-Form search page results
- From the Find-A-Form detail page

Developer notes:
These two page types contain vastly different code surrounding the download links, despite performing the same activity, which is to display a PDF Guidance modal (or immediately download the asset if javascript is disabled).

The main cause this code differs to the degree that it does is because the pages are assembled differently: 

- *The find-a-form search page** is a FE owned page which leverages the Lighthouse Forms API to seed react widgets. 
- **However the detail pages** are built as static CMS owned pages during content-build content release, and the download links are not react components. Instead the page leverages traditional DOM link tags, and javascript is used to perform necessary functions by first listening to the download click events.

- **Form search page result:** https://github.com/department-of-veterans-affairs/vets-website/blob/7e75ac2f212fd90f1a42bf998ea702c20f1c025e/src/applications/static-pages/find-forms/components/SearchResult.jsx#L196
- github**Form detail page download widget:** https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/static-pages/find-forms/download-widget

## Acceptance Criteria
- [x] Determine whether the Find Forms search results and detail pages can share the same code for the `Download VA Form XXX` link. If the code can be shared:
  - [x] Implement the changes to do so
  - [x] Update any unit or e2e tests that are affected
  - [x] Run the test plan that comes from [this ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20280)
- [ ] If the code cannot be shared between the two pages, document the reasoning in this ticket

**Comment by FranECross** [comment]
@randimays Thanks, Randi! Do you want this turned into a Spike, with a subsequent (do the work) ticket?

**Comment by randimays** [comment]
@FranECross I don't think we need to split it up - the research/implementation will go hand-in-hand on this one.

**Comment by FranECross** [comment]
@randimays Thanks so much for spelunking and the explanation. I totally agree with your assessment and path forward, and the need for design input. I wonder if Dave could pop into and out of the ticket to provide insight/input?

**Comment by randimays** [comment]
Thanks @FranECross - @davidmpickett given the above, does this sound like something you could give quick guidance on, or need a separate ticket to think through?

**Comment by davidmpickett** [comment]
So the question is basically where to display the error message on the search results listing page? My instinct would be to mirror the behavior on the detail page and have the alert replace the download link. Here's a quick mock up

![Image](https://github.com/user-attachments/assets/28601dfb-f387-4e3e-a272-c5645eb5369d)

Is that doable with the technical constraints? Would you need any more specifics from me?

**Comment by randimays** [comment]
@davidmpickett This seems doable to me. I'll report back if I run into any snags. Thank you!

**Comment by randimays** [comment]
This ticket is awaiting code review from our team (Platform has already approved). Will likely carry over due to not being able to make the prod deploy today.
